lib/charms/layer/apache_bigtop_base.py

```diff
@@ -53,9 +53,14 @@ def get_ip_for_interface(self, network_interface):
 
         @param str network_interface: either the name of the
         interface, or a CIDR range, in which we expect the interface's
-        ip to fall.
+        ip to fall. Also accepts 0.0.0.0 as a special case, which will
+        simply return 0.0.0.0.
```

Comment:
kwmonroe: well, you just expanded my test cases by a lot :-p

But yeah, I think that expanding the number of ways that people might want to say "listen on all interfaces" is a good idea. Will test and push a change.